### ISSUE TITLE
Create `alpine` release of revive compiler (musl.libc)

### ISSUE THREAD

**Issue by smiasojed** [paritytech/revive#45]
Create the release workflow for `revive` compiler.
As artefacts provide:
1. revive-musl version
2. alpine docker
3. ~~revive-glibc version~~

**Comment by xermicus** [comment]
The musl build should be beneficial for the remix backend.

**Comment by athei** [comment]
Glibc version should not be necessary. A statically linked revive-musl should have zero dependencies.

**Comment by athei** [comment]
@wpt967 Can you take care of this and then go back to the debugging work?

**Comment by wpt967** [comment]
Sure.

**Comment by xermicus** [comment]
@wpt967 thanks a lot!

I think how we should do it for now: The musl workflow should only run on pushes to the main branch and depend on the success of [build-ubuntu-x86](https://github.com/paritytech/revive/blob/main/.github/workflows/rust.yml#L13) (see [here](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_run)). This second workflow can then just build musl, build revive linked against musl, and finally build and publish an alpine Docker image containing the musl revive build.

Should probably also just add a Makefile target install-musl that produces a release build of revive linked against musl.

So, essentially each time we push something to main and the CI passes we'll get a fresh image, which should be good to iterate quickly. Later we can think how to do releases properly (we also want some matrix testing against `solc` versions etc., but thats for later).

**Comment by smiasojed** [comment]
In the Docker image, we need to include both 'resolc-musl' and 'solc-musl'. I would like to ask you to create the first revive release, which I can use as the base for the revive-remix-backend.

**Comment by athei** [comment]
I wouldn't add the `musl` suffix to `resolc`. We should not do any dynamically linked releases anyways as they wouldn't be portable. Or is there any reason to do so?

**Comment by smiasojed** [comment]
I just wanted to emphasize that we need both in the musl versions. We shouldn't have any suffix.

**Comment by xermicus** [comment]
https://gitlab.com/taricorp/llvm-sys.rs/-/issues/44

With this we can give it at shot again. Just need to make sure we pull in the latest version through `inkwell`. Probably our best bet is on using https://github.com/rust-cross/rust-musl-cross as cross-compilation ought to work even with proc-macros.

**Comment by athei** [comment]
So there is an issue with llvm-system which is fixed on master but no release, yet?

**Comment by xermicus** [comment]
No, sorry I didn't realize it. They [backported](https://gitlab.com/taricorp/llvm-sys.rs/-/commits/llvm-18.0?ref_type=heads) the fix to llvm-sys `181.2.0` which is published. So probably only a matter of bumping it in inkwell

**Comment by xermicus** [comment]
Fully static builds work fine now. See the example `Dockerfile` in #154

**Comment by xermicus** [comment]
We don't want to release docker containers: `resolc` depends on `solc`. It's not clear what version this docker would ship. But the musl release makes it trivial for people to build their own containers.
